lib/repl.js

```diff
@@ -689,8 +690,31 @@ function intFilter(item) {
   return /^[A-Za-z_$]/.test(item);
 }
 
+const defaultProperties = {
+  ARRAY: Object.getOwnPropertyNames([]).filter(intFilter),
+  BUFFER: Object.getOwnPropertyNames(Buffer.alloc(1)).filter(intFilter)
+};
+
+function mayBeLargeObject(obj) {
+  return (Array.isArray(obj) || Buffer.isBuffer(obj));
+}
+
 function filteredOwnPropertyNames(obj) {
   if (!obj) return [];
+  if (mayBeLargeObject(obj) && obj.length > 1e6) {
+    this._writeToOutput('\r\n');
+    process.emitWarning(
+      'Instance is too large so the completion may missing some custom ' +
+      'properties.',
+      'REPLWarning',
+      undefined,
+      undefined,
+      true);
```

Comment:
@aqrln The point is:

```js
> var ele = Buffer.alloc(1);
> ele.biu = 1;
> ...
> ele.<tab>
...

ele.biu

> var ele = Buffer.alloc(1e6 + 1);
> ele.biu = 1;
> ...
> ele.<tab>
...
```

The developers may be confused that why there's no `biu`, that completion is what truly they want to find.